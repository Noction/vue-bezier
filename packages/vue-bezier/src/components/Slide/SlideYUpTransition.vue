<template>
  <component
    :is="componentType"
    :tag="tag"
    type="animation"
    v-bind="{ ...$attrs, ...hooks }"
    enter-active-class="slide-y-in"
    move-class="slide-move"
    leave-active-class="slide-y-out"
  >
    <slot />
  </component>
</template>

<script lang="ts">
export const customProps = {
  styles: {
    default: () => {
      return {
        animationFillMode: 'both',
        animationTimingFunction: 'cubic-bezier(.25,.8,.50,1)'
      }
    },
    type: Object
  }
}

export default {
  inheritAttrs: false
}
</script>

<script setup lang="ts">
import {
  buildComponentType,
  buildEmits,
  buildHooks,
  buildProps,
  buildTag
} from '../../composable'

const props = defineProps(buildProps(customProps))
const emit = defineEmits(buildEmits())

const componentType = buildComponentType(props)
const tag = buildTag(props)
const hooks = buildHooks(props, emit)

</script>

<style lang="scss">
  @import 'move';

  @keyframes slide-y-in {
    from {
      opacity: 0;
      transform: translateY(-15px);
    }

    to { opacity: 1; }
  }

  @keyframes slide-y-out {
    from { opacity: 1; }

    to {
      opacity: 0;
      transform: translateY(-15px);
    }
  }

  .slide-y-in { animation-name: slide-y-in; }
  .slide-y-out { animation-name: slide-y-out; }
</style>
