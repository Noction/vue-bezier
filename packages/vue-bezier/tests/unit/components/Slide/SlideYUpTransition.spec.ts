import SlideYUpTransition from '../../../../src/components/Slide/SlideYUpTransition.vue'
import { mount } from '@vue/test-utils'

describe('components / Slide / SlideYUpTransition', () => {
  it('When mounting component with default props, it should work fine', () => {
    const wrapper = mount(SlideYUpTransition)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('When mounting component with group enabled prop, it should work fine', () => {
    const wrapper = mount(SlideYUpTransition, { props: { group: true } })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
