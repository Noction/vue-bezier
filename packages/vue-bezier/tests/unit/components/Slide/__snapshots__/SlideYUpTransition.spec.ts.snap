// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components / Slide / SlideYUpTransition > When mounting component with default props, it should work fine 1`] = `"<transition-stub type="animation" enteractiveclass="slide-y-in" leaveactiveclass="slide-y-out" appear="false" persisted="false" css="true" tag="span" move-class="slide-move"></transition-stub>"`;

exports[`components / Slide / SlideYUpTransition > When mounting component with group enabled prop, it should work fine 1`] = `"<transition-group-stub tag="span" type="animation" enteractiveclass="slide-y-in" moveclass="slide-move" leaveactiveclass="slide-y-out" appear="false" persisted="false" css="true"></transition-group-stub>"`;
